src/snowflake/snowpark/dataframe.py

```diff
@@ -664,14 +664,15 @@ def _execute_and_get_query_id(
         self, *, statement_params: Optional[Dict[str, str]] = None
     ) -> str:
         """This method is only used in stored procedures."""
-        return self._session._conn.get_result_query_id(
-            self._plan,
-            _statement_params=create_or_update_statement_params_with_query_tag(
-                statement_params or self._statement_params,
-                self._session.query_tag,
-                SKIP_LEVELS_THREE,
-            ),
-        )
+        with open_telemetry_context_manager(self._execute_and_get_query_id, self):
```

Comment:
dumb question: how we pick the apis to add tracing to?

I see we have apis like `to_pandas_batches` which also performs `return self._session._conn.execute(` but we don't collect tracing of this API.

Reply:
At this point, we only trace certain action function in dataframe and all udf/stored proc registration process. We may add more in future